src/backend/config/deployments.py

```diff
@@ -1,140 +1,35 @@
-from backend.schemas.deployment import Deployment
 from backend.services.logger.utils import LoggerFactory
 
 logger = LoggerFactory().get_logger()
 
 
-class ModelDeploymentName(StrEnum):
-    CoherePlatform = "Cohere Platform"
-    SageMaker = "SageMaker"
-    Azure = "Azure"
-    Bedrock = "Bedrock"
-    SingleContainer = "Single Container"
-
-
-use_community_features = Settings().get('feature_flags.use_community_features')
+ALL_MODEL_DEPLOYMENTS = { d.name(): d for d in BaseDeployment.__subclasses__() }
 
-# TODO names in the map below should not be the display names but ids
-ALL_MODEL_DEPLOYMENTS = {
-    ModelDeploymentName.CoherePlatform: Deployment(
-        id="cohere_platform",
-        name=ModelDeploymentName.CoherePlatform,
-        deployment_class=CohereDeployment,
-        models=CohereDeployment.list_models(),
-        is_available=CohereDeployment.is_available(),
-        env_vars=COHERE_ENV_VARS,
-    ),
-    ModelDeploymentName.SingleContainer: Deployment(
-        id="single_container",
-        name=ModelDeploymentName.SingleContainer,
-        deployment_class=SingleContainerDeployment,
-        models=SingleContainerDeployment.list_models(),
-        is_available=SingleContainerDeployment.is_available(),
-        env_vars=SC_ENV_VARS,
-    ),
-    ModelDeploymentName.SageMaker: Deployment(
-        id="sagemaker",
-        name=ModelDeploymentName.SageMaker,
-        deployment_class=SageMakerDeployment,
-        models=SageMakerDeployment.list_models(),
-        is_available=SageMakerDeployment.is_available(),
-        env_vars=SAGE_MAKER_ENV_VARS,
-    ),
-    ModelDeploymentName.Azure: Deployment(
-        id="azure",
-        name=ModelDeploymentName.Azure,
-        deployment_class=AzureDeployment,
-        models=AzureDeployment.list_models(),
-        is_available=AzureDeployment.is_available(),
-        env_vars=AZURE_ENV_VARS,
-    ),
-    ModelDeploymentName.Bedrock: Deployment(
-        id="bedrock",
-        name=ModelDeploymentName.Bedrock,
-        deployment_class=BedrockDeployment,
-        models=BedrockDeployment.list_models(),
-        is_available=BedrockDeployment.is_available(),
-        env_vars=BEDROCK_ENV_VARS,
-    ),
-}
 
+def get_installed_deployments() -> list[type[BaseDeployment]]:
+    installed_deployments = list(ALL_MODEL_DEPLOYMENTS.values())
 
-def get_available_deployments() -> dict[ModelDeploymentName, Deployment]:
-    if use_community_features:
+    if Settings().safe_lookup("feature_flags", "use_community_features"):
         try:
             from community.config.deployments import (
                 AVAILABLE_MODEL_DEPLOYMENTS as COMMUNITY_DEPLOYMENTS_SETUP,
             )
-
-            model_deployments = ALL_MODEL_DEPLOYMENTS.copy()
-            model_deployments.update(COMMUNITY_DEPLOYMENTS_SETUP)
-            return model_deployments
-        except ImportError:
+            installed_deployments = [*installed_deployments, *COMMUNITY_DEPLOYMENTS_SETUP.values()]
```

Comment:
Suggestion: use `installed_deployments.extend([other list])`, this should modify it in place